host/python/src/one_sdk/client.py

```diff
@@ -0,0 +1,160 @@
+from typing import Any, Mapping, Optional, Self
+
+import os
+import os.path
+
+from one_sdk.app import WasiApp, SecurityValuesMap
+from one_sdk.error import UnexpectedError
+from one_sdk.platforrm import PythonFilesystem, PythonNetwork, PythonPersistence
+
+CORE_PATH = os.path.abspath(os.path.join(__file__, "../../../assets/core.wasm"))
+if "CORE_PATH" in os.environ:
+	CORE_PATH = os.environ["CORE_PATH"]
+
+class InternalClient:
+	def __init__(
+		self,
+		assets_path: str,
+		token: Optional[str],
+		superface_api_url: str
+	):
+		self._assets_path = assets_path
+		self._core_path = CORE_PATH
+		self._ready = False
+		self._app = WasiApp(
+			PythonFilesystem(),
+			PythonNetwork(),
+			PythonPersistence(token, superface_api_url)
+		)
+
+	def resolve_profile_url(self, profile: str) -> str:
+		resolved_profile = profile.replace('/', '.')
+		path = os.path.abspath(os.path.join(self._assets_path, f"{resolved_profile}.profile"))
+
+		return f"file://{path}"
+
+	def resolve_map_url(self, profile: str, provider: str) -> str:
+		resolved_profile = profile.replace('/', '.')
+		path = os.path.abspath(os.path.join(self._assets_path, f"{resolved_profile}.{provider}.map.js"))
+
+		return f"file://{path}"
+	
+	def resolve_provider_url(self, provider: str) -> str:
+		path = os.path.abspath(os.path.join(self._assets_path, f"{provider}.provider.json"))
+
+		return f"file://{path}"
+	
+	def init(self, core_path = None):
+		if self._ready:
+			return
+		
+		with open(self._core_path, "rb") as file:
+			self._app.load_core(file.read())
+		self._app.init()
+		self._ready = True
+	
+	def destroy(self):
+		if not self._ready:
+			return
+		
+		self._app.destroy()
+		self._ready = False
+	
+	def perform(
+		self,
+		profile: str,
+		provider: str,
+		usecase: str,
+		input: Any,
+		parameters: Mapping[str, str] = {},
+		security: Optional[SecurityValuesMap] = None
+	) -> Any:
+		if security is None:
+			security = dict()
+
+		self.init()
+
+		profile_url = self.resolve_profile_url(profile)
+		provider_url = self.resolve_provider_url(provider)
+		map_url = self.resolve_map_url(profile, provider)
+		
+		try:
+			return self._app.perform(
+				profile_url = profile_url,
+				provider_url = provider_url,
+				map_url = map_url,
+				usecase = usecase,
+				input = input,
+				parameters = parameters,
+				security = security
+			)
+		except UnexpectedError as e:
+			if e.name == "WebAssemblyRuntimeError":
+				self.destroy()
+				self.init()
+			raise
+	
+	def send_metrics(self):
+		self._app.send_metrics()
+
+class UseCase:
+	def __init__(self, internal: InternalClient, profile: "Profile", name: str):
+		self._internal = internal
+		self._profile = profile
+		self.name = name
+
+	def perform(
+		self,
+		input: Any,
+		provider: str,
+		parameters: Mapping[str, str] = {},
+		security: Optional[Mapping[str, Mapping[str, str]]] = None
```

Comment:
Assume these are flattened, because in python there isn't nice way to have options argement as in JS?

Reply:
It's actually the javascript way that is not nice in my opinion. The issue in JS is that if you have too many arguments it tends to be hard to keep track of which one is which. Python solves this with keyword arguments. Plus, as you say, it would be quite cumbersome to not have them flattened.

I think users should always be encouraged to use these with the keyword. If we want, we could even force this by using the asterisk (`def fn(a, b, *, kw_only1, kw_only2)`) syntax.

Reply:
Ok, let's keep named arguments.